### PR TITLE
Enhanced api pages (Part-I)

### PR DESCRIPTION
**Pull request by IIITM-Jay**
## Description
This PR deals with enhancing the ```api``` pages on the respective latest  ***Manual*** release of the corresponding *4.12* version of the *OCaml*.  Basically, in this part of the modification with respect to the ```api```, the bottom navigation has been added so as to provide the desired ease and comfortability to visit its various pages.  

## Issues
1. The ***Bottom Navigation*** is missing from the ```api``` pages.
2. Consequently, it becomes difficult to navigate through the next and the previous pages.
3. *Copyright* section is also absent at the footer.
4. The ***Index of keywords*** page has the option to redirect to the ```Home``` instead of ```The Ocaml API``` page.
5. *Bottom Navigation* stylings are also not included in the ```style.css``` in order to give the corresponding stylings to the same.

## Currently displaying pages
*The Following is the screen-capture of one of the following pages of the ```api```:* 

Screenshot from 2021-03-13 19-48-03

***It should be clearly noted that the bottom navigations are not present***

## Changes Incorporated
- [x] Added the ***previous*** and the ***next*** page navigations
- [x] Incorporate the *Copyright* section.
- [x] Made appropriate and necessary changes to the ***Index of Keywords*** to navigate to the corresponding ```The OCaml API``` page.
- [x] Accurate navigations are added to the last page of the ```api``` i.e., ```Index of Module Types``` page to correctly navigate through the ```The OCaml Compiler API```.
- [x] Added bottom navigation stylings in the ```style.css``` file.

## Resultant Output as Expected
The incorporated changes are highlighted as follows in the following :
 
Webp net-gifmaker(1)

### CODE REVIEW

**Comment by IIITM-Jay** [comment]
Hi, @patricoferris May I request to have feedback and suggestions on the above commit.
waiting for your valuable guidance on the same...

Meanwhile, I have tested this PR on the local build so many times before creating it and I am perfectly getting the desired output. I have also checked the codes that I have added, but not able to get any particular issue or something wrong. I don't know why this build is getting failed... any help from your side...

**Comment by IIITM-Jay** [comment]
@patricoferris firstly I thought that it might be the uneven spacing error or  simply the indentation error... but after rectifying those, ```Ocurrent deployer``` still shows the same error as below. The error shows ```OS package installation failed```, this makes me to :thinking: think of a specific build issue, any of the ***package missing problem*** ( I must guess) What's your opinion...  :pray: 


```E: Failed to fetch http://security.debian.org/debian-security/pool/updates/main/o/openssl/libssl-dev_1.1.1d-0+deb10u3_amd64.deb  404  Not Found [IP: 151.101.38.132 80]
E: Failed to fetch http://deb.debian.org/debian/pool/main/libx/libxml2/libxml2_2.9.4+dfsg1-7+b3_amd64.deb  404  Not Found [IP: 151.101.38.132 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
OS package installation failed

**Comment by patricoferris** [comment]
Hi @IIITM-Jay,

Really appreciate your enthusiasm and effort :)) Two points:

 - I think the reason it is failing is because your branch isn't rebased on `master` and specifically [this commit](https://github.com/ocaml/ocaml.org/commit/d1ec5804066cc56e8d93e8a6ce48173b2e2e1d16).
 - I'm not completely sure, but I think the [manual pages](https://github.com/ocaml/ocaml/tree/trunk/manual) are generated (post-processed) using a specific tool (see [this related issue](https://github.com/ocaml/ocaml.org/issues/1257)) so any changes made directly to the generated HTML output will not be seen in later releases of the manual. I'm of the opinion it would be better to add those changes upstream on the tool rather than patching the output here, but maybe I'm completely wrong about the process.
 
Also (it's completely up to you) feel free to open issues ahead of submitting PRs for changes -- a quick discussion can be useful to make sure your hard-work is focused in the right area/solving the real problem.

Thanks again for your contributions.

**Comment by IIITM-Jay** [comment]
Thank you @patricoferris for your valuable suggestions and feedback. Yeah, now I realized my mistakes too.
You are absolutely right for the things you have discussed here. The manual generating site is converting ```LaTeX to html``` and I have a sound knowledge and hands-on experience with LaTeX too and yes I would love to work on it soon.

But I have one question, whether working on it will be regarded as a part of any of outreachy projects like the one is ***improving the ocaml website***.

**Comment by Octachron** [comment]
Commenting here as a mockup: 
* Linking the various indexes with previous and next links seems to be of dubious value: the indexes are not related. 
* Same as linking the compiler libraries from the standard libraries: the two set of libraries are not related.

File: site/releases/4.12/htmlman/manual067.html [hunk 1]
```diff
@@ -98,9 +98,9 @@ <h1 class="chapter" id="sec580">Index of keywords</h1>
 </li></ul></td></tr>
 </tbody></table>
 </div>
-<div class="bottom-navigation"><a class="previous" href="old.html">« Chapter ‍29 Recently removed or moved libraries (Graphics, Bigarray, Num, LablTk)</a><a class="next up" href="index.html">Home</a></div>
+<div class="bottom-navigation"><a class="previous" href="old.html">« Chapter ‍29 Recently removed or moved libraries</a><a class="next up" href="../api/index.html">The OCaml API »</a></div>
```
Comment (Octachron) [inline site/releases/4.12/htmlman/manual067.html:101]: This change is wrong.

File: site/releases/4.12/api/index.html [hunk 2]
```diff
@@ -1,453 +0,0 @@
...
-</td></tr>
-<tr><td class="module"><a href="Weak.html">Weak</a></td><td><div class="info">
-<p>Arrays of weak pointers and hash sets of weak pointers.</p>
-
-</div>
-</td></tr>
-</tbody></table>
-
-
-</div></body></html>
```
Comment (Octachron) [inline site/releases/4.12/api/index.html:453]: Why are you deleting this file?

**Comment by Octachron** [comment]
I think it is fine to discuss change on the generated html files here before transferring those changes to the upstream manual post-processing.

File: site/releases/4.12/htmlman/manual067.html (hunk 1 above)
Comment (IIITM-Jay) [reply site/releases/4.12/htmlman/manual067.html:101]: @Octachron Actually, I have just shorten the previous chapter name as when we write ***The OCaml API*** as the next page navigation, it will become little crowdy and hence I thought to just shorten it by removing the bracket contents, otherwise i am absolutely sorry for the above commit.

File: site/releases/4.12/api/index.html (hunk 2 above)
Comment (IIITM-Jay) [reply site/releases/4.12/api/index.html:453]: No, no it's deleted by mistakenly in my last commit, it's there in my local computer and if i will find the right time to do PR at the other repo, at that time, I will take care of this.

**Comment by IIITM-Jay** [comment]
> Commenting here as a mockup:
>
>      Linking the various indexes with previous and next links seems to be of dubious value: the indexes are not related.

@Octachron  Yes, I understood the point. My main motive to do this is to only make the readers feel easy and comfortable while navigating to the next and previous pages.

**Comment by Octachron** [comment]
Except that they are not the next/previous pages in any meaningful way: readers don't read an index from top to bottom and then go to the next index.

File: site/releases/4.12/api/index.html (hunk 2 above)
Comment (Octachron) [reply site/releases/4.12/api/index.html:453]: Note that the translation to upstream will not be straightforward. This is why I am encouraging you to finalize the targeted html feature before moving the PR to ocaml/ocaml.

Comment (IIITM-Jay) [reply site/releases/4.12/api/index.html:453]: yeah, I know, If I am not wrong, the post processed manual is using ```LaTeX to HTML``` conversion. Moreover, I have a sound knowledge and hands-on experience with LaTeX too and hence I thought to try this out.

**Comment by IIITM-Jay** [comment]
> Except that they are not the next/previous pages in any meaningful way: readers don't read an index from top to bottom and then go to the next index.

yeah, now I am realizing that these are general indexes and not the next and previous pages.

Hence, I think this is not a valid issue to be worked upon...
waiting for your valuable suggestions and feedback on the same.

**Comment by IIITM-Jay** [comment]
@Octachron  Based on my last comment, if don't have any problem, may I close this PR... as I am not finding it worth and meaningful too to be worked on.

**Comment by Octachron** [comment]
You own this PR, you can close it whenever you want. (You may want to wait for comments from other people, I am far from being an authority in term of web design).

**Comment by IIITM-Jay** [comment]
Closing the PR as this contains the work related to ***Manual*** and hence the same to be done at this [branch](https://github.com/ocaml/ocaml/tree/trunk/manual)

**Comment by patricoferris** [comment]
Thanks @Octachron for providing a review and feedback, I think I agree with everything you have said (I too am no authority on web design).

@IIITM-Jay I think it would make sense to close this PR (you just did whilst I was writing this), the commits are a little hard to follow and I'm struggling to easily identify what the goal is. This is why I suggested a better workflow is to create an issue where discussion can be had before changing code. Issues are more light-weight and consume less resources for you. Perhaps this could be incorporated into the CONTRIBUTING.md (i.e. a workflow for suggesting changes), I'll add that on my todo list.

If you do have more thoughts on improvements to the site (or the manual) please do open an issue. Note that I think we decided the ocaml.org repo is a good place to try out different stylings/content of the manual as it only requires changes to HTML/CSS -- with agreement we could then look into upstreaming them to the compiler.
